ray-operator/controllers/ray/batchscheduler/yunikorn/yunikorn_scheduler.go

```diff
@@ -49,65 +59,157 @@ func (y *YuniKornScheduler) DoBatchSchedulingOnSubmission(_ context.Context, _ *
+	if value, exist := rayJob.Labels[sourceKey]; exist {
+		y.logger.Info("Updating submitter pod template label based on RayJob labels",
+			"sourceKey", sourceKey, "targetKey", targetKey, "value", value)
+		submitterTemplate.Labels[targetKey] = value
+	}
+}
+
+// AddMetadataToChildResourceFromRayCluster adds essential labels and annotations to the Ray pods
 // the yunikorn scheduler needs these labels and annotations in order to do the scheduling properly
-func (y *YuniKornScheduler) AddMetadataToPod(ctx context.Context, app *rayv1.RayCluster, groupName string, pod *corev1.Pod) {
+func (y *YuniKornScheduler) AddMetadataToChildResourceFromRayCluster(ctx context.Context, rayCluster *rayv1.RayCluster, groupName string, pod *corev1.Pod) {
 	// the applicationID and queue name must be provided in the labels
-	y.populatePodLabels(ctx, app, pod, RayClusterApplicationIDLabelName, YuniKornPodApplicationIDLabelName)
-	y.populatePodLabels(ctx, app, pod, RayClusterQueueLabelName, YuniKornPodQueueLabelName)
+	y.populatePodLabelsFromRayCluster(ctx, rayCluster, pod, RayClusterApplicationIDLabelName, YuniKornPodApplicationIDLabelName)
+	y.populatePodLabelsFromRayCluster(ctx, rayCluster, pod, RayClusterQueueLabelName, YuniKornPodQueueLabelName)
 	pod.Spec.SchedulerName = y.Name()
 
 	// when gang scheduling is enabled, extra annotations need to be added to all pods
-	if y.isGangSchedulingEnabled(app) {
+	if y.isGangSchedulingEnabled(rayCluster) {
 		// populate the taskGroups info to each pod
-		y.populateTaskGroupsAnnotationToPod(ctx, app, pod)
+		y.populateTaskGroupsAnnotationToPod(ctx, rayCluster, pod)
 
 		// set the task group name based on the head or worker group name
 		// the group name for the head and each of the worker group should be different
 		pod.Annotations[YuniKornTaskGroupNameAnnotationName] = groupName
 	}
 }
 
-func (y *YuniKornScheduler) isGangSchedulingEnabled(app *rayv1.RayCluster) bool {
-	_, exist := app.Labels[utils.RayClusterGangSchedulingEnabled]
-	return exist
+func (y *YuniKornScheduler) AddMetadataToChildResourceFromRayJob(ctx context.Context, rayJob *rayv1.RayJob, rayCluster *rayv1.RayCluster, submitterTemplate *corev1.PodTemplateSpec) {
+	// the applicationID and queue name must be provided in the labels
+	y.populateRayClusterLabelsFromRayJob(ctx, rayJob, rayCluster, RayClusterApplicationIDLabelName, YuniKornPodApplicationIDLabelName)
+	y.populateRayClusterLabelsFromRayJob(ctx, rayJob, rayCluster, RayClusterQueueLabelName, YuniKornPodQueueLabelName)
+	if submitterTemplate.Labels == nil {
+		submitterTemplate.Labels = make(map[string]string)
+	}
+	y.populateSubmitterPodTemplateLabelsFromRayJob(ctx, rayJob, submitterTemplate, RayClusterApplicationIDLabelName, YuniKornPodApplicationIDLabelName)
+	y.populateSubmitterPodTemplateLabelsFromRayJob(ctx, rayJob, submitterTemplate, RayClusterQueueLabelName, YuniKornPodQueueLabelName)
+
```

Comment:
```suggestion
	y.populateSubmitterPodTemplateLabelsFromRayJob(ctx, rayJob, submitterTemplate, RayClusterQueueLabelName, YuniKornPodQueueLabelName)
    submitterTemplate.Spec.SchedulerName = y.Name()
```

I think we need to set `SchedulerName` to ensure the submitter Job is scheduled by `YuniKorn` instead of the default kube-scheduler. 
Same as head pod and worker pod.

<img width="1022" height="711" alt="image" src="https://github.com/user-attachments/assets/92ff3b34-791d-4212-9cf8-8c15be1ab3bd" />
Ref: https://yunikorn.apache.org/docs/developer_guide/deployment/

Reply:
Added, thanks a lot 🙏